### PR TITLE
Shade hive-service-rpc and thrift in Spark engine

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

The spark sql engine should use the same hive-service-rpc version with that of kyuubi server to prevent conflicts with spark native hive service rpc library.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
Please also update NOTICE of spark engine module

File: externals/kyuubi-spark-sql-engine/pom.xml [hunk 1]
```diff
@@ -219,6 +221,20 @@
...
+                            <shadedPattern>${kyuubi.shade.packageName}.org.apache.hive</shadedPattern>
+                            <includes>
+                                <include>org.apache.hive.**</include>
+                            </includes>
+                        </relocation>
+                        <relocation>
+                            <pattern>org.apache.thrift</pattern>
+                            <shadedPattern>${kyuubi.shade.packageName}.org.apache.thrift</shadedPattern>
+                            <includes>
+                                <include>org.apache.thrift.**</include>
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/pom.xml:235]: fb303 has another package name

Comment (turboFei) [reply externals/kyuubi-spark-sql-engine/pom.xml:235]: what is that?

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/pom.xml:235]: Something related with thrift https://mvnrepository.com/artifact/org.apache.hive/hive-service-rpc/3.1.2

**Review by pan3793 — APPROVED** [review]
LGTM

**Comment by turboFei** [comment]
thanks, merging to master

**Pull request closed without merging**